File: solutions/3555. Smallest Subarray to Sort in Every Sliding Window/3555.py
class Solution:
  def minSubarraySort(self, nums: list[int], k):
    ans = []

    for i in range(len(nums) - k + 1):
      window = nums[i:i+k]
      sortedWindow = sorted(window)
      l = 0
      r = k - 1
      while l < k and window[l] == sortedWindow[l]:
        l += 1
      while r >= 0 and window[r] == sortedWindow[r]:
        r -= 1
      ans.append(0 if l > r else r - l + 1)

    return ans

File: solutions/3555. Smallest Subarray to Sort in Every Sliding Window/3555.cpp
class Solution {
 public:
  vector<int> minSubarraySort(vector<int>& nums, int k) {
    const int n = nums.size();
    vector<int> ans;

    for (int i = 0; i <= n - k; ++i) {
      vector<int> window(nums.begin() + i, nums.begin() + i + k);
      vector<int> sortedWindow = window;
      ranges::sort(sortedWindow);
      int l = 0;
      int r = k - 1;
      while (l < k && window[l] == sortedWindow[l])
        ++l;
      while (r >= 0 && window[r] == sortedWindow[r])
        --r;
      ans.push_back(l > r ? 0 : r - l + 1);
    }

    return ans;
  }
};
